server/src-lib/Hasura/Server/Init.hs

```diff
@@ -492,6 +502,12 @@ parseServerPort = optional $
          help (snd servePortEnv)
        )
 
+parseServerHost :: Parser (Maybe String)
```

Comment:
Change the type to `Parser (Maybe HostPreference)`. `HostPreference` is an instance of `IsString` type class so this should work without any other change.